src/components/timeline/StackGraph.js

```diff
@@ -153,33 +160,48 @@ class StackGraph extends PureComponent<Props> {
       }
       const height = callNodeTable.depth[callNodeIndex] * yPixelsPerDepth;
       const xPos = (sampleTime - range[0]) * xPixelsPerMs;
+      let samplesBucket;
       if (hasSelectedCallNodePrefix(callNodeIndex)) {
-        highlightedSamples.height.push(height);
-        highlightedSamples.xPos.push(xPos);
+        samplesBucket = highlightedSamples;
       } else {
-        regularSamples.height.push(height);
-        regularSamples.xPos.push(xPos);
+        const stackIndex = ensureExists(
+          thread.samples.stack[i],
+          'A stack must exist for this sample, since a callNodeIndex exists.'
+        );
+        const categoryIndex = thread.stackTable.category[stackIndex];
+        const category = categories[categoryIndex];
+        if (category.name === 'Idle') {
+          samplesBucket = idleSamples;
+        } else {
+          samplesBucket = regularSamples;
+        }
       }
+      samplesBucket.height.push(height);
+      samplesBucket.xPos.push(xPos);
       nextMinTime = sampleTime + minGapMs;
     }
 
     // Draw the regular samples first, and then the highlighted samples.
     // This means that we only set ctx.fillStyle twice, which saves on time
     // that's spent parsing color strings.
```

Comment:
Nit: This comment is out of date. Perhaps it can be deleted as it is pretty clear from the code what's happening, except maybe the time saving part.

Reply:
Thanks, nice catch! I updated it.